server/mediaserver.go

```diff
@@ -819,18 +830,47 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 			"Content-Disposition": {"attachment; filename=" + fname},
 			"Rendition-Name":      {profile},
 		}
-		fw, err := mw.CreatePart(hdrs)
+		fw, err = mw.CreatePart(hdrs)
 		if err != nil {
 			glog.Error("Could not create multipart part ", err)
 			break
 		}
 		if len(renditionData[i]) > 0 {
-			io.Copy(fw, bytes.NewBuffer(renditionData[i]))
+			_, err = io.Copy(fw, bytes.NewBuffer(renditionData[i]))
+			if err != nil {
+				break
+			}
 		} else {
-			fw.Write([]byte(url))
+			_, err = fw.Write([]byte(url))
+			if err != nil {
+				break
+			}
 		}
 	}
-	mw.Close()
+	if err == nil {
+		err = mw.Close()
+	}
+	if err != nil {
+		glog.Errorf("Error sending transcoded response url=%s err=%v", r.URL.String(), err)
+		if monitor.Enabled {
+			monitor.HTTPClientTimedOut2()
+		}
+		return
+	}
+	w.(http.Flusher).Flush()
```

Comment:
Think its good practice (based on the [http.Flusher](https://golang.org/pkg/net/http/#Flusher) docs), to make sure that the `http.ResponseWriter` implements the `http.Flusher` interface i.e.

```golang
if f, ok := w.(http.Flusher); ok {
    f.Flush()
}
```

Reply:
fixed in e5d86dda116caadf2c6a7d99a94ce2bb63e4215f